src/utils/sources-tree/addToTree.js

```diff
@@ -82,12 +82,22 @@ function findOrCreateNode(
  * walk the source tree to the final node for a given url,
  * adding new nodes along the way
  */
-function traverseTree(url: Object, tree: Node, debuggeeHost: ?string) {
+function traverseTree(
+  url: Object,
+  tree: Node,
+  debuggeeHost: ?string,
+  projectRoot: string
+) {
   url.path = decodeURIComponent(url.path);
 
   const parts = url.path.split("/").filter(p => p !== "");
   parts.unshift(url.group);
 
+  if (projectRoot) {
+    const rootParts = projectRoot.replace("://", "").split("/");
+    parts.splice(0, rootParts.length - 2);
```

Comment:
this logic looks really similar to the other logic. What's going on here?

Reply:
If the project root is set, let's say `/example.com/components`, then it's path is `/components`
so we just keep the last part (I am thinking of making it easy to grasp, maybe use can use `url` library)

Now we start the tree path with /components.

Reply:
Got it